src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Sources/ManualResetValueTaskSourceCore.cs

```diff
@@ -211,47 +211,68 @@ private void SignalCompletion()
             {
                 if (_executionContext != null)
                 {
-                    ExecutionContext.RunInternal(
-                        _executionContext,
-                        (ref ManualResetValueTaskSourceCore<TResult> s) => s.InvokeContinuation(),
-                        ref this);
+                    // ExecutionContext flow enabled
+                    Thread currentThread = Thread.CurrentThread;
+                    // Restore the ExecutionContext before executing anything.
+                    ExecutionContext? currentContext = ExecutionContext.RestoreInternal(_executionContext, currentThread);
+
+                    ExceptionDispatchInfo? edi = null;
+                    if (_capturedContext == null)
+                    {
+                        // Default scheduler
+                        if (RunContinuationsAsynchronously)
+                        {
+                            ThreadPool.QueueUserWorkItem(_continuation, _continuationState, preferLocal: true);
```

Comment:
UnsafeQueueUserWorkItem?

Reply:
This is the captured `ExecutionContext` path; the other branch of the if/else `else if (_capturedContext == null)` uses `UnsafeQueueUserWorkItem` (i.e. where `_executionContext == null`)

Reply:
Ah, thanks.